tests/Transport/AwsAuthV4Test.php

```diff
@@ -205,4 +205,35 @@ public function testSignsWithEnvironmentalCredentials(): void
             );
         }
     }
+
+    /**
+     * @group unit
+     * @depends testSignsWithProvidedCredentials
+     */
+    public function testStripsTrailingDotInHost(): void
+    {
+        $host = $this->_getHost();
+        $hostWithTrailingDot = $host.'.';
+
+        $config = [
+            'persistent' => false,
+            'transport' => 'AwsAuthV4',
+            'aws_access_key_id' => 'foo',
+            'aws_secret_access_key' => 'bar',
+            'aws_session_token' => 'baz',
+            'aws_region' => 'us-east-1',
+            'host' => $hostWithTrailingDot,
+        ];
+        $client = $this->_getClient($config);
+
+        try {
+            $client->request('_stats');
+        } catch (GuzzleException $e) {
+            $guzzleException = $e->getGuzzleException();
```

Comment:
Can we move this out of the `catch`?

We should make sure that the exception is thrown and according to our expectations.
This test case is not covering the possibility that no exceptions are thrown from `$client->request('_stats');`

WDYT?

Reply:
To be honest we just replicated what we already found in the file (see all other tests in the file above). Somehow those changes must've made their way to master as well. But don't get me wrong, you have a point there - it's just maybe not in the scope of our change to rework this and make it better within our PR?